File: c/TypeScript/Symbol.ts
import { TypeClassification } from "./TypeClassification.js";

export class Symbol {
    public name: string = "";
    public classification: Set<TypeClassification> = new Set();
    public members: Map<string, Symbol> = new Map();
    public parent: Symbol | null = null;
    public predefined: boolean = false;
    public definedFile: string = "";
    public definedLine: number = 0;
    public definedColumn: number = 0;

    public toString(): string {
        let result = this.name;
        const classificationStr = Array.from(this.classification)
            .map(c => TypeClassification[c])
            .join(", ");
        result += " (with classification " + classificationStr + ")";
        if (this.definedFile && this.definedFile.length > 0) {
            result += " defined at " + this.definedFile + ":" + this.definedLine + ":" + this.definedColumn;
        }
        return result;
    }
}

File: c/Python3/Symbol.py
from TypeClassification import TypeClassification

class Symbol:
    def __init__(self):
        self.name = ""
        self.classification = set()
        self.members = dict()
        self.parent = None
        self.predefined = False
        self.definedFile = ""
        self.definedLine = 0
        self.definedColumn = 0

    def __str__(self):
        result = self.name
        classification_str = ", ".join(TypeClassification(c).name for c in self.classification)
        result += " (with classification " + classification_str + ")"
        if self.definedFile and len(self.definedFile) > 0:
            result += " defined at " + self.definedFile + ":" + str(self.definedLine) + ":" + str(self.definedColumn)
        return result
